infrastructure/containers/postgres/automatic-backup/backup_or_restore.py

```diff
@@ -85,32 +152,64 @@ def backup_database_on_interval(seconds):
     )
     while True:
         time.sleep(seconds)
-        backup_database(make_backup_fpath())
+        s3_backup_object_name = make_backup_object_name_from_datetime()
+        backup_database(s3_backup_object_name)
 
 
 ###################
 # --- Restore --- #
 ###################
 
 
-def get_most_recent_backup_fpath():
+def download_backup_object(
+    session: boto3.session.Session, backup_bucket_name: str, backup_obj_name: str, backup_fpath: str
+):
+    """
+    Downloads the object_name backup from the backup_bucket_name S3 bucket.
+
+    Params:
+        session (boto3.session.Session): the AWS session to be used for downloading the file
+        backup_bucket_name (str): the name of the bucket to downlaod the file from
+        backup_obj_name (str): the name of the object to download
+        backup_fpath (str): the filepath for the downloaded file
+    """
+    s3_client = session.client("s3")
+    s3_client.download_file(backup_bucket_name, backup_obj_name, backup_fpath)
+
+
+def list_bucket_objects(session: boto3.session.Session, backup_bucket_name: str) -> List[str]:
+    """
+    Returns a list of all objects in the backup_bucket_name S3 bucket.
+
+    Params:
+        session (boto3.session.Session): the AWS session to be used for downloading the file
+        backup_bucket_name (str): the name of the bucket to list the objects in
+
+    Returns:
+        (List[str]): a list containing the names of all of the objects in the bucket
+    """
+    bucket = session.resource("s3").Bucket(backup_bucket_name)
+    return [obj.key for obj in bucket.objects.all()]
+
+
+def get_most_recent_backup_obj_name(session: boto3.session.Session) -> str:
+    """Returns the name of the most recent backup in S3."""
     # get a list of all the backup files
-    backup_files = BACKUP_DIR.glob("*.sql.gz")
+    backup_files = list_bucket_objects(session, BACKUP_BUCKET)
 
     # get the most recent backup file
-    get_datetime_from_fpath = lambda fpath: datetime.strptime(str(fpath.name), FILENAME_DATETIME_FORMAT)
+    get_datetime_from_fpath = lambda fpath: datetime.strptime(fpath, FILENAME_DATETIME_FORMAT)
```

Comment:
Hah, nice.

Reply:
A downside about using `lambda` functions is that they don't show up in auto-generated documentation.